pkg/machines/components/nicEdit.jsx

```diff
@@ -53,10 +53,26 @@ class EditNICModal extends React.Component {
     constructor(props) {
         super(props);
 
+        let defaultNetworkSource;
+        let currentSource;
+        let availableSources = [];
+        if (props.network.type === "network") {
+            availableSources = props.networks.map(network => network.name);
+            currentSource = props.network.source.network;
+        } else if (props.network.type === "direct") {
+            availableSources = props.nodeDevices;
+            currentSource = props.network.source.device;
```

Comment:
Watch out for a typo, this should be `props.network.source.dev`

Reply:
Done.